rmw_fastrtps_shared_cpp/src/init_rmw_context_impl.cpp

```diff
@@ -102,6 +102,7 @@ rmw_fastrtps_shared_cpp::decrement_context_impl_ref_count(rmw_context_t * contex
     delete common_context;
     context->impl->common = nullptr;
     context->impl->participant_info = nullptr;
+    return ret;
   }
   return RMW_RET_OK;
```

Comment:
One way I think we could make this a bit clearer is to do a slight refactor of this method to move the `context->impl->count != 0` check to the top of the method.  That is, the start of the method would look something like:

```
std::lock_guard<std::mutex> guard(context->impl->mutex);
context->impl->count--;
if (context->impl->count > 0) {
  return RMW_RET_OK;
}

// implicitly, the count is now 0, and we can do cleanup things.
```

(we would still need your fix to `return ret` here, but now I think it would be a bit more clear that you have to do so, and the lack of it would have been picked up by normal compiler flags)

Reply:
I love using early returns whenever possible. Fixed.